packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -128,6 +249,7 @@ function findRoutesAndCreateOptimalPath(
                         .integerValue()
                         .toNumber(),
                 );
+                memo.isVip = vipSourcesSet.has(sample.source);
```

Comment:
This should be the same for all samples in the path right? Maybe move it to the last arg to `reduce()` with optional chaining on element `0`?

Reply:
Nice optimization 🙏